python/shotgun_model/shotgunmodel.py

```diff
@@ -422,17 +433,42 @@ def _load_data(self, entity_type, filters, hierarchy, fields, order=None, seed=N
         #
         # the reason these are split up is because the params tend to be constant and
         # the filters keep varying depending on user input.
-
+        #
+        # some comment regarding the fields that make up the hash
+        #
+        # fields, order, hierarchy are all coming from Shotgun
+        # and are used to uniquely identify the cache file. Typically,
+        # code using the shotgun model will keep these fields constant
+        # while varying filters. With the filters hashed separately,
+        # this typically generates a folder structure where there is one
+        # top level folder containing a series of cache files
+        # all for different filters.
+        #
+        # the schema generation is used for advanced implementations 
+        # See constructor docstring for details.
+        # 
+        # bg_load_thumbs is hashed so that the system can cache 
+        # thumb and non-thumb caches independently. This is because
+        # as soon as you start caching thumbnails, qpixmap will be used
+        # internally by the serialization and this means that you get 
+        # warnings if you try to use those caches in threads. By keeping
+        # caches separate, there is no risk that a thumb cache 'pollutes'
+        # a non-thumb cache.
```

Comment:
no no - in that case, you would run the model with thumbnails enabled (set in the constructor of the model), then the view would decide whether to show thumbs or not. the model would keep the thumbs around all the time.